frontend/javascripts/oxalis/controller/camera_controller.ts

```diff
@@ -225,6 +226,7 @@ export function rotate3DViewTo(id: OrthoView, animate: boolean = true): void {
   // Use width and height to keep the same zoom.
   let width = tdCamera.right - tdCamera.left;
   let height = tdCamera.top - tdCamera.bottom;
+  let far = tdCamera.far;
```

Comment:
I'm wondering why this value is recomputed every time the `rotate3DViewTo` function is called. Isn't this value static and can be computed once when initializing the cameras (or once the dataset extent is known)?

Reply:
Also note that there is already the following line in `componentDidMount` of the `CameraController`:

```
const far = 8000000;
```

This line should be removed and probably replaced with the new calculation.

On another note: Could `far` simply be set to `Number.MAX_VALUE`? Or does something speak against this?

Reply:
Yeah, moving the code to `componentDidMount` in `CameraController` sounds good 👍 .

> Could far simply be set to Number.MAX_VALUE? Or does something speak against this?

Honestly, I cannot think of an argument against this. As far as I know, we want to render everything within the scene of the 3d viewport, right? (At least we do not want to clip scene elements that are far in the back; only those who are cut off on the left, right, upper, down side)
What do you think @daniel-wer?

Should I do the `const far = Number.MAX_VALUE;` change for every ortho viewport or only for the 3d viewport. I think it should not affect the other viewports as well.

Reply:
> Honestly, I cannot think of an argument against this. As far as I know, we want to render everything within the scene of the 3d viewport, right? (At least we do not want to clip scene elements that are far in the back; only those who are cut off on the left, right, upper, down side)
What do you think @daniel-wer?

There could be issues with number inaccuracies or performance implications, but that should be testable.

> Should I do the const far = Number.MAX_VALUE; change for every ortho viewport or only for the 3d viewport. I think it should not affect the other viewports as well.

I would say that only makes sense for the 3d viewport since in the other viewports the plane is offset by the clipping distance, to make nodes that would be behind the plane visible, and everything further away is blocked by the plane. I'm not sure how the maximum value for the clipping distance is chosen and/or how the far value for the orthogonal viewports is chosen, though. I don't think it makes sense to set far to a number larger than the maximum current clipping distance (+ a potential offset due to the camera position, not sure).

Reply:
> There could be issues with number inaccuracies or performance implications, but that should be testable.

How would you test for this? Create an extremely large dataset?

Reply:
Ok, using ``Number.MAX_VALUE` is not an option. The gif below shows the rendering bugs I got by using that for the 3d viewport as a value for the far plane.

![output](https://user-images.githubusercontent.com/39529669/170029633-d27aeb05-fb58-44f3-9168-bccf2d3e127a.gif)

Thus I'll use the heuristic as done previously but now setting it only once in the `componentDidMount`.

Reply:
Thanks for giving it a try!